presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java

```diff
@@ -1406,21 +1400,22 @@ private static String formatHash(Optional<Symbol>... hashes)
         return "[" + Joiner.on(", ").join(symbols) + "]";
     }
 
-    private static String formatFrame(WindowFrame frame)
+    private static String formatFrame(WindowNode.Frame frame)
     {
         StringBuilder builder = new StringBuilder(frame.getType().toString());
-        FrameBound start = frame.getStart();
-        if (start.getValue().isPresent()) {
-            builder.append(" ").append(start.getOriginalValue().get());
+
+        Optional<Symbol> startValue = frame.getStartValue();
+        if (startValue.isPresent()) {
+            builder.append(" ").append(frame.getOriginalStartValue().get());
```

Comment:
You have a contract `startValue.isPresent() ⇒ originalStartValue().isPresent()`. Can you enforce this in `WindowNode.Frame` ctor?